docs/specs/moniker.yml

```diff
@@ -240,3 +240,164 @@ outputs:
+    }
+  218c13d0/docs/a.json: |
+    {
+      "content": "<p>Moniker: netcore-1.1</p>",
+      "monikers": ["netcore-1.1"]
+    }
+  4667fedf/docs/a.json: |
+    {
+      "content": "<p>Moniker: netcore-2.0</p>",
+      "monikers": ["netcore-2.0"]
+    }
+---
+# Define same uid with multiple moniker ranges, 
+# reference to it without moniker in the same docset, shoule take the latest one
+inputs:
+  docfx.yml: |
+    monikerRange:
+      'docs/v1/**/*.md': '< netcore-2.0'
+      'docs/v2/**/*.md': '>= netcore-2.0'
+    routes:
+      docs/v1/: docs/
+      docs/v2/: docs/
+    monikerDefinition: monikerDefinition.json
+  docs/v1/a.md: |
+    ---
+    uid: a
+    title: netcore-1.1
+    monikerRange: '> netcore-1.0'
+    ---
+    Moniker: netcore-1.1
+  docs/v2/a.md: |
+    ---
+    uid: a
+    title: netcore-2.0
+    monikerRange: '<= netcore-2.0'
+    ---
+    Moniker: netcore-2.0
+  docs/a.md: |
+    No version
+  docs/b.md: |
+    Link to @a
+  monikerDefinition.json: |
+    {
+      "monikers": [
+        { "name": "netcore-1.0", "product": ".NET Core" },
+        { "name": "netcore-1.1", "product": ".NET Core" },
+        { "name": "netcore-2.0", "product": ".NET Core" },
+        { "name": "netcore-2.1", "product": ".NET Core" }
+      ]
+    }
+outputs:
+  docs/b.json: |
+    {
+      "content": "<p>Link to <a href=\"docs/a.json\">netcore-2.0</a></p>",
+      "monikers": [] 
+    }
+  docs/a.json: |
+    {
+      "content": "<p>No version</p>",
+      "monikers": []
+    }
+  218c13d0/docs/a.json: |
+    {
+      "content": "<p>Moniker: netcore-1.1</p>",
+      "monikers": ["netcore-1.1"]
+    }
+  4667fedf/docs/a.json: |
+    {
+      "content": "<p>Moniker: netcore-2.0</p>",
+      "monikers": ["netcore-2.0"]
+    }
+---
+# Define same uid with overlapping moniker range
+inputs:
+  docfx.yml: |
+    monikerRange:
+      'docs/v1/**/*.md': '<= netcore-2.0'
+      'docs/v2/**/*.md': '>= netcore-2.0'
+    routes:
+      docs/v1/: docs/
+      docs/v2/: docs/
+    monikerDefinition: monikerDefinition.json
+  docs/v1/a.md: |
+    ---
+    uid: a
+    title: netcore-1.1, netcore-2.0
+    monikerRange: '> netcore-1.0'
+    ---
+    Moniker: netcore-1.1
+  docs/v2/a.md: |
+    ---
+    uid: a
+    title: netcore-2.0
+    monikerRange: '> netcore-1.0'
+    ---
+    Moniker: netcore-2.0
+  docs/c.md: Link to @a
+  monikerDefinition.json: |
+    {
+      "monikers": [
+        { "name": "netcore-1.0", "product": ".NET Core" },
+        { "name": "netcore-1.1", "product": ".NET Core" },
+        { "name": "netcore-2.0", "product": ".NET Core" },
+        { "name": "netcore-2.1", "product": ".NET Core" }
+      ]
+    }
+outputs:
+  docs/c.json: |
+    {"content":"<p>Link to @a</p>"}
+  build.log: |
+    ["error","moniker-overlapping","Two or more documents have defined overlapping moniker range ('netcore-2.0', 'netcore-2.1') ('netcore-1.1', 'netcore-2.0')"]
+    ["info","at-uid-not-found","Cannot find uid 'a' using xref '@a'","docs/c.md"]
```

Comment:
>info [](start = 6, length = 4)

warning?

Reply:
Since `@` is not only used for xref reference, better not log a warning for it.